src/server/req_delete.c

```diff
@@ -407,6 +407,9 @@ req_deletejob(struct batch_request *preq)
 		if ((i == JOB_STATE_EXITING) && (forcedel == 0)) {
 			req_reject(PBSE_BADSTATE, 0, preq);
 			return;
+		} else if (i == JOB_STATE_EXPIRED){
+			req_reject(PBSE_NOHISTARRAYSUBJOB, 0, preq);
+			return;
```

Comment:
Is it better to do the check in svr_saveorpurge_finjobhist() function?

Reply:
It will be difficult as we wont be having the subjob state in check_deletehistoryjob()